### PR TITLE
Fixes from e2e tests

### PR DESCRIPTION
**Pull request by paolobueno**
## Motivation

https://issues.jboss.org/browse/AEROGEAR-2218
https://issues.jboss.org/browse/AEROGEAR-2266

## Description

Readd the type and name fields to metrics data.

## Progress

- [x] Add `getDisplayName`
- [x] Rename `getName` to `getType`

### CODE REVIEW

File: auth/src/main/java/org/aerogear/mobile/security/SecurityCheckType.java [hunk 1]
```diff
@@ -72,9 +72,9 @@ public SecurityCheck getSecurityCheck() {
     /**
      * Gets the name of the security check.
```
Comment (darahayes) [inline auth/src/main/java/org/aerogear/mobile/security/SecurityCheckType.java]: comment rot - Please update this.

File: auth/src/main/java/org/aerogear/mobile/security/SecurityCheckType.java [hunk 2]
```diff
@@ -72,9 +72,9 @@ public SecurityCheck getSecurityCheck() {
...
      *
-     * @return {@link String} {@link SecurityCheck#getName()}
+     * @return {@link String} {@link SecurityCheck#getType()}
      */
     public String getName() {
```
Comment (darahayes) [inline auth/src/main/java/org/aerogear/mobile/security/SecurityCheckType.java]: This seems confusing to me. It's a method called getName() that internally calls getType()

File: auth/src/main/java/org/aerogear/mobile/security/AsyncSecurityCheckExecutor.java [hunk 3]
```diff
@@ -129,7 +129,7 @@ public AsyncSecurityCheckExecutor build() {
         final AtomicInteger count = new AtomicInteger(checks.size());
 
         for (final SecurityCheck check : checks) {
-            res.put(check.getName(), (executorService.submit(() -> {
+            res.put(check.getDisplayName(), (executorService.submit(() -> {
```
Comment (ziccardi) [inline auth/src/main/java/org/aerogear/mobile/security/AsyncSecurityCheckExecutor.java]: Use getType instead of getDisplayName

Comment (wei-lee) [reply auth/src/main/java/org/aerogear/mobile/security/AsyncSecurityCheckExecutor.java]: +1. The display name is not guaranteed to be unique, but type is.

**Comment by paolobueno** [comment]
@ziccardi do you have any other suggestions for the method names? like getName()` and `getDisplayName()`...

**Comment by darahayes** [comment]
@ziccardi from a maintainer/contributor perspective is also unintuitive to have a map with the key of `type` and the value returned from `getName()`. This is even worse now because there is also a `name` key in the same map. This would be really confusing.

The key prop should map directly to the name of the accessor method and to the actual value, i.e. `{type: check.getType(), name: check.getName()}` I feel like not adhering to this will definitely come back to bite us later on.

I also agree that `getType()` might seem unintuitive to an end user of the SDK. 

The class name e.g. `org.aerogear.security.foo` is being used as a unique identifier for checks. , perhaps something like `getId()` would more appropriate.

For example

```
{id: check.getId(), name: check.getName()} //pseudocode
```

which results in:

```
{id: "org.aerogear.security.EmulatorCheck", name: "Emulator Check"}
```

What do you think?

**Comment by darahayes** [comment]
I'd like one of the following:

### 1. The current PR which does the following

```
{type: check.getType(), name: check.getName()}
```

resulting in

```
{type: "org.aerogear.security.EmulatorCheck", name: "Emulator Check"}
```

This does not require any changes on the server side.

### 2. Changes Paolo suggested:

```
{name: check.getName(), displayName: check.getDisplayName()}
```

resulting in

```
{name: "org.aerogear.security.EmulatorCheck", displayName: "Emulator Check"}
```

### 3. Another suggestion:

```
{id: check.getId(), name: check.getName()} //pseudocode
```

which results in:

```
{id: "org.aerogear.security.EmulatorCheck", name: "Emulator Check"}
```

As long as the key name matches the method name I'm fairly happy. Options 2 and 3 would require a minor change on the server side but it's tiny.

**Comment by wei-lee** [comment]
I prefer option 1 or option 3. Don't think we need to explicitly call it `displayName`. Option 3 might be a little bit better as value is indeed used as an id. Also calling it `id` usually implies it should be unique.

**Comment by ziccardi** [comment]
@darahayes I believe that the 1st one is confusing for the end user.
I don't like the 2nd one

If I should choose, I would go for the 3rd one

**Comment by darahayes** [comment]
I prefer option 3 also. I will submit an appropriate PR against the app metrics server in the next few minutes

File: auth/src/main/java/org/aerogear/mobile/security/metrics/SecurityCheckResultMetric.java [hunk 4]
```diff
@@ -24,7 +24,7 @@
     private final String TAG = "SecurityCheckResultMetric";
 
     public static final String IDENTIFIER = "security";
-    public static final String KEY_TYPE = "type";
+    public static final String KEY_ID = "ID";
```
Comment (darahayes) [inline auth/src/main/java/org/aerogear/mobile/security/metrics/SecurityCheckResultMetric.java]: can this be `id`? all other keys are lowercase

File: auth/src/main/java/org/aerogear/mobile/security/impl/SecurityCheckResultImpl.java [hunk 5]
```diff
@@ -26,8 +26,8 @@
      */
     public SecurityCheckResultImpl(@NonNull SecurityCheck check, final boolean passed) {
         SecurityCheck nonNullCheck = nonNull(check, "check");
-        this.type = nonNullCheck.getType();
-        this.displayName = nonNullCheck.getDisplayName();
+        this.type = nonNullCheck.getId();
+        this.displayName = nonNullCheck.getName();
```
Comment (darahayes) [inline auth/src/main/java/org/aerogear/mobile/security/impl/SecurityCheckResultImpl.java]: is `this.type` and `this.displayName` confusing?

**Comment by darahayes** [comment]
@ziccardi @paolobueno @wei-lee please see associated PR for the server side implementation. I've tagged you guys for approval.

**Pull request merged**